### ISSUE TITLE
Firewall policy with internet service doesn't add specified service under Destination field

### ISSUE THREAD

**Issue by kamlesh-mutha** [fortinet-ansible-dev/ansible-galaxy-fortios-collection#87]
We are using ansible task (pasted below) to create firewall policy which would open access from our internal subnet to Microsoft Office365 service. It does create firewall policy but doesn't add selected service in Destination field. I can manually update that same policy to add internet service from console though.

Ansible Task:

> 
> - name: Configuring firewall policy from app-internal to outlook internet service
>   fortios_firewall_policy:
>     vdom: "{{ fortinet_vdom }}"
>     https: "True"
>     ssl_verify: "False"
>     host: "{{ firewall_host }}"
>     username: "{{ firewall_username }}"
>     password: "{{ firewall_password }}"
>     firewall_policy:
>       state: "present"
>       name: "internal-subnet-to-office365"
>       srcintf:
>         -
>           name: "port3"
>       srcaddr:
>         -
>           name: "internet-subnet"
>       dstintf:
>         -
>           name: "port1"
>       schedule: "always"
>       internet_service: "enable"
>       internet_service_id:
>         -
>           id:  "327782"
>       action: "accept"
>       policyid: "11"
>       nat: "enable"
>       ssl_ssh_profile: "certificate-inspection"

Ansible version: 2.9.7
Fortiosapi module: 1.0.5
Fortigate Version: 6.4.3 

Are we missing something here?

**Comment by mbdraks** [comment]
You should be using the Ansible Collection v1.1.7 for FOS 6.4, with this collection you don't use the ID but the name.

Here's a working playbook:

```yaml
#!/usr/bin/env ansible-playbook
---
- hosts: Enterprise_Core

  vars:
    firewall_policy_list:
      - policyid: 0
        name: O365
        srcintf:
          - name: port2
        dstintf:
          - name: port3
        srcaddr:
          - name: all
        internet_service: enable
        internet_service_name:
          - name: Microsoft-Office365
        action: accept
        schedule: always
        logtraffic: all

  tasks:
    - name: Create Firewall Policy
      fortinet.fortios.fortios_firewall_policy:
        state: present
        access_token: "{{ fortios_access_token }}"
        firewall_policy:
          policyid: "{{ item.policyid }}"
          name: "{{ item.name }}"
          srcintf: "{{ item.srcintf }}"
          dstintf: "{{ item.dstintf }}"
          srcaddr: "{{ item.srcaddr }}"
          internet_service: "{{ item.internet_service }}"
          internet_service_name: "{{ item.internet_service_name }}"
          action: "{{ item.action }}"
          schedule: "{{ item.schedule }}"
          logtraffic: "{{ item.logtraffic }}"
      loop: "{{ firewall_policy_list }}"
```

**Comment by hkatyal-spi** [comment]
when we have tried with Ansible Collection v1.1.7 it's giving an error for the internet_service_name
"msg": "Unsupported parameters for (fortios_firewall_policy) module: internet_service_name found in firewall_policy.

**Comment by chillancezen** [comment]
hi @hkatyal-spi ,

just as Michel said, 

`internet_service_id` is not available since 6.4, `internet_service_name ` is the available one.
```
    ...
    internet_service_name:
       - name: serviceA
       - name: serviceB
       - name: serviceC
    ....
```

could you please share your playbook for  with us one more time?

**Comment by chillancezen** [comment]
mark `closed` for now, please feel free to reopen if the issue persists.
